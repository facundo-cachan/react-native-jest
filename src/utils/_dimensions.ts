import { Dimensions, PixelRatio } from 'react-native'

let { height, width } = Dimensions.get('screen')

/**
 * Converts provided width percentage to independent pixel (dp).
 * @param  {string} widthPercent The percentage of screen's width that UI element should cover
 *                               along with the percentage symbol (%).
 * @return {number}              The calculated dp depending on current device's screen width.
 */
const widthPercentageToDP = (widthPercent: string | number) => {
  // Parse string percentage input and convert it to number.
  const elemWidth =
    typeof widthPercent === 'number' ? widthPercent : parseFloat(widthPercent)
  // Use PixelRatio.roundToNearestPixel method in order to round the layout
  // size (dp) to the nearest one that correspons to an integer number of pixels.
  return PixelRatio.roundToNearestPixel((width * elemWidth) / 100)
}

/**
 * Converts provided height percentage to independent pixel (dp).
 * @param  {string} heightPercent The percentage of screen's height that UI element should cover
 *                                along with the percentage symbol (%).
 * @return {number}               The calculated dp depending on current device's screen height.
 */
const heightPercentageToDP = (heightPercent: string | number) => {
  // Parse string percentage input and convert it to number.
  const elemHeight =
    typeof heightPercent === 'number'
      ? heightPercent
      : parseFloat(heightPercent)

  // Use PixelRatio.roundToNearestPixel method in order to round the layout
  // size (dp) to the nearest one that correspons to an integer number of pixels.
  return PixelRatio.roundToNearestPixel((height * elemHeight) / 100)
}

/*
* Implement that piece of code to have orientation detection

Dimensions.get('screen').width < Dimensions.get('screen').height ? 'portrait' : 'landscape'
*/

export { height, width, widthPercentageToDP as wp, heightPercentageToDP as hp }
