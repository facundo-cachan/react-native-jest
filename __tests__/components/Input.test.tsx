import { beforeEach, describe, expect, test } from '@jest/globals';
import { screen } from '@testing-library/react-native';

import { toJSON } from '../utils/customRender';

import { Inputs } from '@components';

import type { InputProps } from '@components';

const props: InputProps = {
  accessibilityLabel: 'asdasdasd',
  style: { backgroundColor: 'red' },

  // secureTextEntry: true,
  value: 'Input Value',
  placeholder: 'PlaceHolder Value'

  // type: 'numeric',
};

describe('Text component', () => {
  beforeEach(() => {
    toJSON(<Inputs.Default {...props} />);
  });
  test('Render component', () => {
    expect(screen).toMatchSnapshot();
  });
  test('Visiblity & text value', () => {
    const input = screen.getByDisplayValue(props.value as string);
    expect(input).toBeOnTheScreen();
    expect(input).toBeVisible();
    expect(input.props.value).toBe(props.value);
  });
  test('Visiblity & placeholder value', () => {
    const input = screen.getByPlaceholderText(props.placeholder as string);
    expect(input).toBeOnTheScreen();
    expect(input).toBeVisible();
    expect(input.props.placeholder).toBe(props.placeholder);
  });
  test('Visiblity & styles values', () => {
    const input = screen.getByDisplayValue(props.value as string);
    expect(input).toBeOnTheScreen();
    expect(input).toBeVisible();
    expect(input.props.style).toContain(props.style);
  });
});
