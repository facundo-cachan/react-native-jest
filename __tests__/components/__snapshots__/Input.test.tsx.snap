// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Render component 1`] = `
<TextInput
  accessibilityLabel="asdasdasd"
  keyboardType="default"
  placeholder="PlaceHolder Value"
  style={
    [
      {
        "fontFamily": "Cochin",
      },
      {
        "backgroundColor": "red",
      },
    ]
  }
  value="Input Value"
/>
`;
